src/commands/send-many-memo-safe.ts

```diff
@@ -0,0 +1,284 @@
+import { Command, flags } from '@oclif/command';
+import { sendMany, Recipient, isNormalInteger, getAddress } from '../builder';
+import {
+  StacksMocknet,
+  StacksMainnet,
+  StacksTestnet,
+  StacksNetwork,
+} from '@stacks/network';
+import {
+  broadcastTransaction,
+  ChainID,
+  validateStacksAddress,
+} from '@stacks/transactions';
+import { STXPostCondition } from '@stacks/transactions/dist/transactions/src/postcondition';
+import fetch, { Response } from 'node-fetch';
+
+type NetworkString = 'mocknet' | 'mainnet' | 'testnet';
+
+const DEFAULT_TESTNET_CONTRACT =
+  'STR8P3RD1EHA8AA37ERSSSZSWKS9T2GYQFGXNA4C.send-many-memo';
+const DEFAULT_MAINNET_CONTRACT =
+  'SP3FBR2AGK5H9QBDH3EEN6DF8EK8JY7RX8QJ5SVTE.send-many-memo';
+
+function uniqueFetchSession() {
+  return (() => {
+    let register: { [key: string]: Promise<Response> } = {};
+    return (url: string) => register[url] || (register[url] = fetch(url));
+  })();
+}
+
+async function checkMemoExpected(
+  network: StacksNetwork,
+  recipients: Recipient[]
+): Promise<string[]> {
+  const fetch = uniqueFetchSession();
+  const results = await Promise.all(
+    recipients
+      .filter(recipient => recipient.memo?.length === 0)
+      .map(recipient =>
+        fetch(
+          network.getAbiApiUrl(recipient.address, 'memo-expected')
+        ).then(response => ({ response, recipient }))
+      )
+  );
+  return results
+    .filter(entry => entry.response.status === 200)
```

Comment:
Do we want to throw an exception here if `entry.response.status` is a 500 error?  Otherwise, we could have a false negative on our hands -- a recipient expected a memo, but this `.filter()` didn't report it.

Reply:
`fetch()` will throw in this case.